### DISCUSSION TITLE
faustesp32 and polyphony

### DISCUSSION THREAD

**Discussion by goofy2k** [grame-cncm/faust#643] [Q&A]
Hi, I have a TTGO TAudio board and am experimenting with DSP. I managed to get the  sawtooth example running. The next step is to try an example that uses SRAM, such as described in your web pages on Faust for ESP32.  The next next step is to try **polyphony**. I have already tried to generate .cpp and .h files in the online IDE, with the number of voices selected to 4. At a first glance I do not see a parameter to select a voice number......   How can I select a specific voice in my ultimate firmware? I do not find a description of that in your project docs or repo.    Thanks, Fred

**Comment by sletz** [comment]
Are you using the `-nvoices <num> ` yes?  Are you using MIDI to control the voices (with the `-midi` option) ?

**Comment by goofy2k** [comment]
I selected polyphony 4 voices with the button in the online IDE. Then
exported the binary. I can include those in my source for the firmware, but
don't know how to use the DSP part in my main app.  A simple example would
help.

Excuse me that we are communicating over several channels :-)  slack, mail,
github. Which one do you prefer?

Fred

Op di 31 aug. 2021 09:19 schreef Stéphane Letz ***@***.***>:
